### PR TITLE
feat: contributing guidelines

### PR DESCRIPTION
**Pull request by nicorikken**
First go at some contributing guidelines / manual. Still a work in progress though.

### CODE REVIEW

File: CONTRIBUTING.adoc [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+=== CoffeeScript lint
+This package uses http://www.coffeelint.org/[CoffeeLint] to ensure that the code is uniformly structured.
+The linter definitions are defined in the link:coffeelint.json[coffeelint.json].
+
+When developing in Atom, the https://atom.io/packages/linter-coffeelint[linter-coffeelint] package exposes the linting in your editor.
+It does require CoffeeLint to be http://www.coffeelint.org/#install[installed].
+
+CoffeeLint can also be triggered from the commandline:
+
+`coffeelint -f coffeelint.json grammars/asciidoc.cson`
```
Comment (ldez) [inline CONTRIBUTING.adoc]: Replace this 2 line by:

```adoc
[source,shell]
----
coffeelint -f coffeelint.json grammars/asciidoc.cson spec/asciidoc-spec.coffee snippets/language-asciidoc.cson
----
```

File: CONTRIBUTING.adoc [hunk 2]
```diff
@@ -0,0 +1,57 @@
+= Contributing
+
+In the spirit of free software, _everyone_ is encouraged to help improve this project.
+
+To contribute code, simply fork the project on GitHub, hack away and send a pull request with your proposed changes.
+
+Feel free to use the https://github.com/asciidoctor/atom-language-asciidoc/issues[issue tracker] to provide feedback or suggestions in other ways.
+
+== Development
+*todo*
```
Comment (ldez) [inline CONTRIBUTING.adoc]: Its personal but I think separate title from section content with a blank line is more readable.

File: CONTRIBUTING.adoc (hunk 1 above)
Comment (nicorikken) [inline CONTRIBUTING.adoc]: Although I'd prefer the first option for cleanness, it happily parses the `node_modules` directory on my machine. And as expected this triggers a load of errors. The latter is better of that reason.

File: CONTRIBUTING.adoc (hunk 2 above)
Comment (nicorikken) [inline CONTRIBUTING.adoc]: Do you mean like:
```
== Development

*todo*
```

Comment (ldez) [inline CONTRIBUTING.adoc]: yes :wink:

**Comment by mojavelinux** [comment]
Thanks for getting this started @nicorikken! This information will definitely help get more people involved.

**Comment by nicorikken** [comment]
Some of my lessons and resources included in the most recent commit. Look forward to some additional comments to improve this, and get to a releasable document.

**Comment by ldez** [comment]
Great job! :+1:

**Comment by nicorikken** [comment]
Just added the test commands offered in issue #50

**Comment by ldez** [comment]
@mojavelinux I have find why `\p{blank}`  works!
It's because Atom use a custom regular expression parser: [oniguruma](https://github.com/kkos/oniguruma)
It support POSIX expression and more!

We can add this refernce to the documentation.

**Comment by mojavelinux** [comment]
Look at how much we learn from documentation! :tada: Amazing.

**Comment by mojavelinux** [comment]
That's also really, really good news because it gives us a lot more flexibility. This guide can help us in a lot of ways. http://www.apeth.com/nonblog/stories/textmatebundle.html We should add it to the contributing guide. wdyt @nicorikken ?

**Comment by ldez** [comment]
The link (http://www.apeth.com/nonblog/stories/textmatebundle.html) is already in the file :wink:

**Comment by mojavelinux** [comment]
Wow, me, get with the program. hahaha ;)

**Comment by mojavelinux** [comment]
I think we should merge this document and use follow-up issues to make improvements. wdyt?

If there's a section you think needs to be completed before we merge, let me know and I'd be glad to help fill in the text.

**Comment by nicorikken** [comment]
I've added the recent regex marks as well, and now merged the whole shebang.

**Comment by mojavelinux** [comment]
:+1:

**Pull request merged**